.github/workflows/ci.yaml

```diff
@@ -32,3 +32,63 @@ jobs:
         run: make local-test
       - name: Run e2e tests
         run: make e2e-tests
+
+  ci-karpenter:
+    runs-on: ubuntu-latest
+    name: ci-karpenter
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.24'
+      - name: Run Gosec Security Scanner
+        uses: securego/gosec@master
+        with:
+          args: -quiet -exclude=G107 ./...
+      - name: Run golangci-lint
+        uses: golangci/golangci-lint-action@v8
+        with:
+          # Optional: version of golangci-lint to use in form of v1.2 or v1.2.3 or `latest` to use the latest version
+          # version: v1.46
+           args: -v --timeout 5m --no-config ./...
+      - name: Install k8s Kind Cluster
+        uses: helm/kind-action@v1.12.0
+        with:
+          install_only: true
+          version: v0.29.0
+      - name: Prepare test environment
+        run: make local-test-karpenter
+      - name: Run e2e tests
+        run: make e2e-tests
+
+  ci-node-labels:
```

Comment:
Wouldn't be a better idea to run these tests as additional steps to the existing github action instead of creating new ones? In this way we avoid duplicating the initial setup